src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -249,10 +249,10 @@ def _prepare(self) -> BaseDistribution:
         return dist
 
     def iter_dependencies(self, with_requires: bool) -> Iterable[Optional[Requirement]]:
+        yield self._factory.make_requires_python_requirement(self.dist.requires_python)
```

Comment:
Maybe add a comment here to explain that the order in which we yield the requirements is important, and putting requires-python first allows us to avoid more expensive checks where possible?

Reply:
I'm not convinced a comment adds much, but it can't hurt. See 5bd1255dd.

Reply:
Thanks - IMO it clarifies the intention so I appreciate you adding it.